### PR TITLE
Feature/wmt 823/allow for batch size of one when calculating workload points

### PR DESCRIPTION
**Pull request by dconey646**

### CODE REVIEW

File: app/services/data/get-app-workloads.js [hunk 1]
```diff
@@ -5,9 +5,7 @@
   const Workload = require('wmt-probation-rules').Workload
   const Locations = require('wmt-probation-rules').Locations
 
-  module.exports = function (initialId, batchSize) {
```
Comment (tswann) [inline app/services/data/get-app-workloads.js]: Any kind of constraint here ensuring that batchSize is > 0?

File: app/services/workers/calculate-workload-points.js [hunk 2]
```diff
@@ -16,11 +16,18 @@ module.exports.execute = function (task) {
   var id = task.additionalData.workloadBatch.startingId
   var batchSize = task.additionalData.workloadBatch.batchSize
   var reportId = task.workloadReportId
+  var maxId = id + batchSize - 1
+
+  if (batchSize === 1) {
```
Comment (tswann) [inline app/services/workers/calculate-workload-points.js]: I think if you set maxId = id above and changed this to batchSize > 1 you could possibly do away with the else which is only being used for logging a variant of the message.

E.g.
```
maxId = id
message = 'single workload message'
if (batchSize > 1) {
 maxId = maxId + batchSize - 1
 append additional stuff to message
}
log(message)

File: app/services/data/get-app-workloads.js (hunk 1 above)
Comment (dconey646) [reply app/services/data/get-app-workloads.js]: Nothing, but we can do a check and throw an error if it isn't.

File: app/services/workers/calculate-workload-points.js [hunk 3]
```diff
@@ -16,11 +16,25 @@ module.exports.execute = function (task) {
   var id = task.additionalData.workloadBatch.startingId
   var batchSize = task.additionalData.workloadBatch.batchSize
   var reportId = task.workloadReportId
+  var maxId = id
+
+  if (batchSize < 0) {
```
Comment (ddebella) [inline app/services/workers/calculate-workload-points.js]: should be `if (batchSize <= 0)`

File: app/services/workers/calculate-workload-points.js [hunk 4]
```diff
@@ -16,11 +16,25 @@ module.exports.execute = function (task) {
...
+    logger.error('Batchsize must be greater than 0')
+    throw (new Error('Batchsize must be greater than 0'))
+  }
+
+  var message = 'Calculating Workload Points for Workloads ' + id + ' - ' + (id + batchSize)
+
+  if (batchSize > 1) {
```
Comment (ddebella) [inline app/services/workers/calculate-workload-points.js]: There is no need to do the check here again if the the above change <=0 is done.

File: app/services/workers/calculate-workload-points.js (hunk 3 above)
Comment (dconey646) [reply app/services/workers/calculate-workload-points.js]: Yep, logic has been reordered

File: app/services/workers/calculate-workload-points.js (hunk 4 above)
Comment (dconey646) [reply app/services/workers/calculate-workload-points.js]: Yea, logic has been reordered

File: test/unit/app/workers/test-calculate-workload-points.js [hunk 5]
```diff
@@ -64,7 +65,7 @@ describe('services/workers/calculate-workload-points', function () {
     getOffenderManagerTypeIdStub.resolves(WORKLOAD_ID)
     getPointsConfigurationStub.resolves({values: pointsHelper.getCaseTypeWeightings(), id: WORKLOAD_ID})
     calculateWorkloadPoints.execute(task).then(function () {
-      expect(getWorkloadsStub.calledWith(WORKLOAD_ID, BATCH_SIZE)).to.equal(true)
+      expect(getWorkloadsStub.calledWith(WORKLOAD_ID, MAX_ID, BATCH_SIZE)).to.equal(true)
       done()
```
Comment (caoimheb) [inline test/unit/app/workers/test-calculate-workload-points.js]: Include a test case for batch sizes of 0, 1, greater than 1 to test new code?

**Pull request merged**